src/note.ts

```diff
@@ -250,10 +496,10 @@ export class Note extends Tickable {
   }
 
   // Get and set the target stave.
-  getStave() {
+  getStave(): Stave | undefined {
     return this.stave;
   }
-  setStave(stave) {
+  setStave(stave: Stave): this {
     this.stave = stave;
     this.setYs([stave.getYForLine(0)]); // Update Y values if the stave is changed.
     this.context = this.stave.context;
```

Comment:
@0xfe this is the reference to context.

Reply:
So if you have `context` as `private`, then you can `this.setContext(this.stave.getContext())`.

Reply:
I have proceeded with your proposal, context is now private. It was protected.